src/main/java/bio/terra/service/cohortbuilder/HintsService.java

```diff
@@ -1,151 +1,124 @@
-                                                    .display("Not Hispanic or Latino")
-                                                    .value(
-                                                        new Literal()
-                                                            .dataType(DataType.INT64)
-                                                            .valueUnion(
-                                                                new LiteralValueUnion()
-                                                                    .int64Val(38003564L))))
-                                            .count(2272403))))))
-        .addDisplayHintsItem(
-            new DisplayHint()
-                .attribute(
-                    new Attribute()
-                        .name("year_of_birth")
-                        .type(Attribute.TypeEnum.SIMPLE)
-                        .dataType(DataType.INT64))
-                .displayHint(
-                    new DisplayHintDisplayHint()
-                        .numericRangeHint(
-                            new DisplayHintNumericRange().min(1909.0D).max(1983.0D))));
+        .sql(sql)
+        .displayHints(
+            displayHints.entrySet().stream()
+                .map(
+                    attrHint -> {
+                      Attribute attr = entity.getAttribute(attrHint.getKey());
+                      DisplayHint hint = attrHint.getValue();
+                      return new bio.terra.model.DisplayHint()
+                          .attribute(ToApiConversionUtils.toApiObject(attr))
+                          .displayHint(hint == null ? null : toApiObject(hint));
+                    })
+                .collect(Collectors.toList()));
+  }
+
+  private DisplayHintDisplayHint toApiObject(DisplayHint displayHint) {
+    switch (displayHint.getType()) {
+      case ENUM -> {
+        EnumVals enumVals = (EnumVals) displayHint;
+        return new DisplayHintDisplayHint()
+            .enumHint(
+                new DisplayHintEnum()
+                    .enumHintValues(
+                        enumVals.getEnumValsList().stream()
+                            .map(
+                                ev ->
+                                    new DisplayHintEnumEnumHintValues()
+                                        .enumVal(
+                                            ToApiConversionUtils.toApiObject(ev.getValueDisplay()))
+                                        .count(Math.toIntExact(ev.getCount())))
+                            .collect(Collectors.toList())));
+      }
+      case RANGE -> {
+        NumericRange numericRange = (NumericRange) displayHint;
+        return new DisplayHintDisplayHint()
+            .numericRangeHint(
+                new DisplayHintNumericRange()
+                    .min(numericRange.getMinVal())
+                    .max(numericRange.getMaxVal()));
+      }
+      default -> throw new SystemException("Unknown display hint type: " + displayHint.getType());
```

Comment:
If you want to, you could move `return` outside the `switch` so it's a switch expression. Then you could remove this `default` since the a switch expression requires exhaustiveness so a missing case would cause a compile error.

Reply:
I think because it declares intermediate variables it can't be done as a switch expression? Or at least, IntelliJ complains if I try. I thought about inlining the casts, but I think that leads to duplicated cast statements for the second code path. What are your thoughts there?